server/middlewares/validateAddress.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable eol-last */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable consistent-return */
+/* eslint-disable indent */
+
+import dotenv from 'dotenv';
+import express from 'express';
+import bodyParser from 'body-parser';
+
+
+const app = express();
+app.use(bodyParser.urlencoded({ extended: true }));
+
+dotenv.load();
+
+class ThirdPartyValidators {
+    isAddressValid(req, res, next) {
+        let { partyAddress } = req.body;
+        partyAddress = partyAddress.trim();
+        const googleMapsClient = require('@google/maps').createClient({
+            key: process.env.GOOGLE_API_KEY,
+            Promise,
+          });
+          googleMapsClient.places({
+              query: partyAddress,
+          })
+            .asPromise()
+            .then((response) => {
+                if (response.json.results) {
+                    req.partyAddr = response.json.results[0].formatted_address;
+                    next();
+                } else {
+                    return res.status(404).send({
```

Comment:
this line of code if unreachable when there is no result from the Google API `res.json.results` becomes an empty array --> `[]` checking if an empty array is defined (if (res.json.results) {...}) returns true so the else case is never captured and instead an error about formatted address pops up. its better to use  `if (res.json.results.length < 1) {}` this will capture the cases where there are no results